packages/contracts/contracts/minter-suite/Minters/MinterSetPriceERC20V5.sol

```diff
@@ -0,0 +1,541 @@
+     * than or equal to the value of project `_project_id`'s maximum invocations that is
+     * set on the core contract.
+     * @dev Note that a `_maxInvocations` of 0 can only be set if the current `invocations`
+     * value is also 0 and this would also set `maxHasBeenInvoked` to true, correctly short-circuiting
+     * this minter's purchase function, avoiding extra gas costs from the core contract's maxInvocations check.
+     * @param _projectId Project ID to set the maximum invocations for.
+     * @param _coreContract Core contract address for the given project.
+     * @param _maxInvocations Maximum invocations to set for the project.
+     */
+    function manuallyLimitProjectMaxInvocations(
+        uint256 _projectId,
+        address _coreContract,
+        uint24 _maxInvocations
+    ) external {
+        _onlyArtist(_projectId, _coreContract);
+        MaxInvocationsLib.manuallyLimitProjectMaxInvocations(
+            _projectId,
+            _coreContract,
+            _maxInvocations,
+            _maxInvocationsProjectConfigMapping[_coreContract][_projectId]
+        );
+        emit ProjectMaxInvocationsLimitUpdated(
+            _projectId,
+            _coreContract,
+            _maxInvocations
+        );
+    }
+
+    /**
+     * @notice Updates this minter's price per token of project `_projectId`
+     * to be '_pricePerTokenInWei`, in Wei.
+     * @dev Note that it is intentionally supported here that the configured
+     * price may be explicitly set to `0`.
+     * @param _projectId Project ID to set the price per token for.
+     * @param _coreContract Core contract address for the given project.
+     * @param _pricePerTokenInWei Price per token to set for the project, in Wei.
+     */
+    function updatePricePerTokenInWei(
+        uint256 _projectId,
+        address _coreContract,
+        uint248 _pricePerTokenInWei
+    ) external {
+        _onlyArtist(_projectId, _coreContract);
+        ProjectConfig storage _projectConfig = _projectConfigMapping[
+            _coreContract
+        ][_projectId];
+        _projectConfig.pricePerTokenInWei = _pricePerTokenInWei;
+        _projectConfig.priceIsConfigured = true;
+        emit PricePerTokenInWeiUpdated(
+            _projectId,
+            _coreContract,
+            _pricePerTokenInWei
+        );
+
+        // sync local max invocations if not initially populated
+        // @dev if local max invocations and maxHasBeenInvoked are both
+        // initial values, we know they have not been populated.
+        MaxInvocationsLib.MaxInvocationsProjectConfig
+            storage _maxInvocationsProjectConfig = _maxInvocationsProjectConfigMapping[
+                _coreContract
+            ][_projectId];
+        if (
+            _maxInvocationsProjectConfig.maxInvocations == 0 &&
+            _maxInvocationsProjectConfig.maxHasBeenInvoked == false
+        ) {
+            syncProjectMaxInvocationsToCore(_projectId, _coreContract);
```

Comment:
I wonder if we want to just always do this – given that `updatePricePerTokenInWei` is an artist-only config method and unconditionally calling `syncProjectMaxInvocationsToCore` makes things more readable and bit safer re: that value's freshness.

Reply:
This was only intended to be an initial sync if the project had never been configured on the minter, and it syncs to the core contract value to be the least restrictive possible.
If we were to sync project max invocations unconditionally, we would overwrite any manually configured, more restrictive max invocations, which we don't want.
That being said, this does feel cluttered - WDYT about moving this to a helper function (with as much logic in the max invocations library as possible)?

Reply:
sounds good - comments updated to be more clear in [6ead59c](https://github.com/ArtBlocks/artblocks-contracts/pull/827/commits/6ead59c6bd66a4c4ce02dac46769013d5b88d377)

Reply:
also updated all new minters to have the same comment (since this is shared functionality across all minters) in [af3d421](https://github.com/ArtBlocks/artblocks-contracts/pull/827/commits/af3d4214431cba9ad189a378f32b76a5b1580d82)

Reply:
TYVM ser 💜 🙏